### PR TITLE
Cleanup content_sao by factorizing similar code parts + little perf fix

### PR DESCRIPTION
**Pull request by nerzhul**
Original PR was #4764

### CODE REVIEW

**Comment by nerzhul** [comment]
@sfan5 @sofar @est31 please look at this simple PR

File: src/content_sao.cpp [hunk 1]
```diff
@@ -920,21 +918,10 @@ std::string PlayerSAO::getClientInitializationData(u16 protocol_version)
 	return os.str();
 }
 
-std::string PlayerSAO::getStaticData()
+void PlayerSAO::getStaticData(std::string *result) const
 {
 	FATAL_ERROR("Deprecated function (?)");
```
Comment (ShadowNinja) [inline src/content_sao.cpp]: `"Deprecated function " FUNCTION_NAME`
And following line is unneeded.

**Comment by ShadowNinja** [comment]
:+1:

File: src/content_sao.cpp (hunk 1 above)
Comment (nerzhul) [inline src/content_sao.cpp]: Macro FATAL_ERROR already set FUNCTION_NAME, but i will remove the (?) and the return
```
  fatal_error_fn((msg), __FILE__, __LINE__, FUNCTION_NAME)
```

**Pull request merged**